File: 백준/Silver/1439. 뒤집기/뒤집기.cc
#include <iostream>

using namespace std;

int main(void) {
    string str;
    int zero = 0, one = 0, idx = 0;
    cin >> str;
    
    while (idx < str.length()) {
        char target = str[idx];
        if (target == '0') zero++;
        else one++;
        while (str[idx] == target) idx++;
    }
    
    cout << min(zero, one) << endl;
        
    return 0;
}

File: 백준/Silver/1439. 뒤집기/뒤집기.py
S = input()

zero, one, idx = 0, 0, 0
while idx < len(S):
    if S[idx] == '0':
        zero += 1
        while idx < len(S) and S[idx] == '0':
            idx += 1
    elif S[idx] == '1':
        one += 1
        while idx < len(S) and S[idx] == '1':
            idx += 1

print(min(zero, one))